### PR TITLE
Avoid wrong filling of  glusterfs Volume Spec at ConstructVolumeSpec()

### PR DESCRIPTION
**Pull request by humblec**
Issue#https://github.com/kubernetes/kubernetes/issues/37625, https://github.com/kubernetes/kubernetes/issues/49108

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -200,10 +200,7 @@ func (plugin *glusterfsPlugin) ConstructVolumeSpec(volumeName, mountPath string)
 	glusterfsVolume := &v1.Volume{
 		Name: volumeName,
 		VolumeSource: v1.VolumeSource{
-			Glusterfs: &v1.GlusterfsVolumeSource{
-				EndpointsName: volumeName,
-				Path:          volumeName,
-			},
+			Glusterfs: &v1.GlusterfsVolumeSource{},
```
Comment (jingxu97) [inline pkg/volume/glusterfs/glusterfs.go]: I think maybe just return error since we cannot reconstruct volume spec for glusterfs.

Comment (humblec) [reply pkg/volume/glusterfs/glusterfs.go]: @jingxu97 I can get the `path` filled by proper volume name , however reconstructing `endpoint` from mount string looks to be impossible. I can return error , hopefully that can cause the reconciler to start unmount process.

**Comment by humblec** [comment]
@jsafrane @jingxu97 Please review.

**Comment by jsafrane** [comment]
It's fine as a short term band aid, however it must be fixed in the long term. Now a volume remains mounted if someone deletes a pod while kubelet is down. These leftover mounts are very hard to debug, as they happen only once in a while, but we've seen them in production and they pollute the machine.

@jingxu97, what do you think?

**Comment by humblec** [comment]
@jsafrane completely agree on we need long term plan to get rid of these circles.  Just pushing this now as a interim fix.

**Comment by jingxu97** [comment]
@jsafrane yes, I agree. We will take some time to fix this whole volume reconstruction logic after 1.9. 
/lgtm

**Comment by humblec** [comment]
@jsafrane @jingxu97 Can you please add 1.9 milestone to this PR ? Thanks!

**Comment by humblec** [comment]
Thanks Jan!

**Pull request merged**